spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/kafka/KafkaProperties.java

```diff
@@ -266,6 +266,13 @@ public Template getTemplate() {
 		 */
 		private Duration heartbeatInterval;
 
+		/**
+		 * Controls how transactional messages are returned when polling the broker
+		 * (non-transactional messages will be unconditionally returned, regardless of
+		 * this setting).
+		 */
+		private String isolationLevel;
```

Comment:
It may make sense if it is mapped as an `org.apache.kafka.common.requests.IsolationLevel` enum type.
Otherwise this one indeed can be mapped as all other generic properties in the `spring.kafka.consumer.properties` container.

More over I think such a property description is too long. Essentially there is just enough to mention that it is mapped onto `ConsumerConfig.ISOLATION_LEVEL_CONFIG)`.

Reply:
Coincedentally, I had the same two thoughts!  On both accounts, I went the direction I did because of the precedent set by existing config options.  I found valid values being defined in metadata instead of enums in [this example](https://github.com/spring-projects/spring-boot/blob/master/spring-boot-project/spring-boot-autoconfigure/src/main/resources/META-INF/additional-spring-configuration-metadata.json#L2184) and I added the description in the comment again based on all of the existing config.

Interestingly, the `isolation.level` property does have its [doc constant](https://github.com/apache/kafka/blob/2.3.0/clients/src/main/java/org/apache/kafka/clients/consumer/ConsumerConfig.java#L270) exposed as `public` even though most other config options are `private`

Reply:
AFAIK, there is no enum for `ConsumerConfig.AUTO_OFFSET_RESET_CONFIG` (certainly not used there) which is why it's handled that way.

The isolation level enum is used in `ConsumerConfig` so it's best to use that.

Reply:
Actually, they have to lower case the enum anyway, so maybe this is the best solution after all.

`IsolationLevel.READ_UNCOMMITTED.toString().toLowerCase(Locale.ROOT);`

Reply:
Still would be better to be based on enum. That way, when they introduce a new level, we are going to be covered automatically together with content assistant in the IDE.